submarineswap_server.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"swapper/mempoolspace"
+	"swapper/submarineswaprpc"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightningnetwork/lnd/zpay32"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Server is a sub-server of the main RPC server.
+type Server struct {
+	ActiveNetParams *chaincfg.Params
+}
+
+// SubSwapServiceInit
+func (s *Server) SubSwapServiceInit(ctx context.Context,
+	in *submarineswaprpc.SubSwapServiceInitRequest) (*submarineswaprpc.SubSwapServiceInitResponse, error) {
+	//Create a new submarine address and associated script
+	addr, script, swapServicePubKey, lockHeight, err := NewSubmarineSwap(
+		s.ActiveNetParams,
+		in.Pubkey,
+		in.Hash,
+	)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[SubSwapServiceInit] addr=%v script=%x pubkey=%x", addr.String(), script, swapServicePubKey)
+	return &submarineswaprpc.SubSwapServiceInitResponse{Address: addr.String(), Pubkey: swapServicePubKey, LockHeight: lockHeight}, nil
+}
+
+// UnspentAmount returns the total amount of the btc received in a watched address
+// and the height of the first transaction sending btc to the address.
+func (s *Server) UnspentAmount(ctx context.Context,
+	in *submarineswaprpc.UnspentAmountRequest) (*submarineswaprpc.UnspentAmountResponse, error) {
+
+	address := in.Address
+	var lockHeight int32
+	addr, _, lh, err := AddressFromHash(s.ActiveNetParams, in.Hash)
+	if err != nil {
+		return nil, err
+	}
+	address = addr.String()
+	lockHeight = int32(lh)
+	c := &mempoolspace.Client{BaseUrl: "https://mempool.space/api"}
```

Comment:
Let's move the mempool space domain to an environment variable and better to create a function in this file getBlockchainClient that returns the mempool space client and use it whenever we need such struct instance.